dali/pipeline/data/copy_to_external.h

```diff
@@ -44,15 +44,41 @@ inline void CopyToExternal(void* dst, AllocType dst_alloc_type,
     use_copy_kernel &= (DstType == AllocType::GPU || DstType == AllocType::Pinned) &&
                        (std::is_same<SrcBackend, GPUBackend>::value || src.is_pinned());
     DeviceGuard d(src.device_id());
-    auto &type_info = TypeTable::GetTypeInfo(DALIDataType::DALI_UINT8);
-    type_info.template Copy<DstBackend, SrcBackend>(dst, src.raw_data(), src.nbytes(), stream,
+    const auto &type_info = src.type();
+    type_info.template Copy<DstBackend, SrcBackend>(dst, src.raw_data(), src.size(), stream,
                                                     use_copy_kernel);
   ), ());  // NOLINT
 
   if (sync)
     CUDA_CALL(cudaStreamSynchronize(stream));
 }
 
+template <typename SrcBackend>
+inline void CopyToExternal(void** dsts, AllocType dst_alloc_type,
+                           const TensorList<SrcBackend> &src,
+                           cudaStream_t stream, bool sync, bool use_copy_kernel) {
```

Comment:
IMHO the `sync` argument is redundant. I'd rather assume that this function is always async and synchronize it externally. Mainly because, when passing `sync=true`, user might be tempted to pass `stream=nullptr`, which might be disastrous